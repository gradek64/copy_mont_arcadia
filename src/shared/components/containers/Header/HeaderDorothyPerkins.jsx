import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { Link } from 'react-router'
import BrandLogo from './../../common/BrandLogo/BrandLogo'
import WishlistHeaderLink from './../../common/WishlistHeaderLink/WishlistHeaderLink'
import ShoppingCart from './../../common/ShoppingCart/ShoppingCart'
import SearchBar from './../../containers/SearchBar/SearchBar'
import AccountIcon from './../../common/AccountIcon/AccountIcon'
import ShippingDestination from './../../common/ShippingDestination/ShippingDestination'
import TopNavMenuBig from '../TopNavMenu/TopNavMenuBig'
import MegaNav from '../MegaNav/MegaNav'

export default class HeaderDorothyPerkins extends Component {
  static propTypes = {
    brandName: PropTypes.string,
    logoVersion: PropTypes.string,
    featureMegaNav: PropTypes.bool,
    sticky: PropTypes.bool,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  renderHeaderIcons = () => {
    const { sticky } = this.props
    const { l } = this.context
    return [
      !sticky && (
        <div key="shipping" className="HeaderDorothyPerkins-shipping">
          <ShippingDestination
            modifier="dorothyPerkinsHeader"
            currencySymbolPosition="right"
            currencySymbolStyle="bracketed"
          />
        </div>
      ),
      <div key="search" className="HeaderDorothyPerkins-search">
        <SearchBar isDesktop />
      </div>,
      !sticky && (
        <div key="account" className="HeaderDorothyPerkins-account">
          <AccountIcon
            popoverMenu
            myAccountText={l`My account`}
            signOutText={l`Sign out`}
          />
        </div>
      ),
      <div key="wishlist" className="HeaderDorothyPerkins-wishlist">
        <WishlistHeaderLink />
      </div>,
      <div key="shoppingCart" className="HeaderDorothyPerkins-shoppingCart">
        <ShoppingCart />
      </div>,
    ]
  }

  render() {
    const { brandName, logoVersion, featureMegaNav, sticky } = this.props
    return (
      <div className={`HeaderDorothyPerkins${sticky ? ' is-sticky' : ''}`}>
        <div className="HeaderDorothyPerkins-brand">
          <Link className="HeaderDorothyPerkins-brandLink" to="/">
            <BrandLogo
              brandName={brandName}
              logoVersion={logoVersion}
              hasResponsive
            />
          </Link>
        </div>
        <div className="HeaderDorothyPerkins-content">
          {!sticky && this.renderHeaderIcons()}
        </div>
        <div className="HeaderDorothyPerkins-navigation">
          {featureMegaNav ? (
            <MegaNav className="HeaderDorothyPerkins-megaNav" />
          ) : (
            <TopNavMenuBig />
          )}
          {sticky && this.renderHeaderIcons()}
        </div>
      </div>
    )
  }
}
