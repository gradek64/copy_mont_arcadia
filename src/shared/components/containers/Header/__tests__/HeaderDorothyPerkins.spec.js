import testComponentHelper from 'test/unit/helpers/test-component'
import HeaderDorothyPerkins from '../HeaderDorothyPerkins'

describe('<HeaderDorothyPerkins />', () => {
  const renderComponent = testComponentHelper(HeaderDorothyPerkins)
  const initialProps = {
    brandName: 'dorothyperkins',
    contentWrapper: 'HeaderBig-content',
    leftCol: 'HeaderBig-left',
    rightCol: 'HeaderBig-right',
    centreCol: 'HeaderBig-centre',
    shoppingBagTotalItems: 3,
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree()).toMatchSnapshot()
    })
  })
})
