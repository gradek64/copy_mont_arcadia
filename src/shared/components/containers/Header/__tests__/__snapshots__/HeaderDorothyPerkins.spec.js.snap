// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderDorothyPerkins /> @renders in default state 1`] = `
<div
  className="HeaderDorothyPerkins"
>
  <div
    className="HeaderDorothyPerkins-brand"
  >
    <Link
      className="HeaderDorothyPerkins-brandLink"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/"
    >
      <BrandLogo
        brandName="dorothyperkins"
        hasResponsive={true}
      />
    </Link>
  </div>
  <div
    className="HeaderDorothyPerkins-content"
  >
    <div
      className="HeaderDorothyPerkins-shipping"
      key="shipping"
    >
      <Connect(ShippingDestination)
        currencySymbolPosition="right"
        currencySymbolStyle="bracketed"
        modifier="dorothyPerkinsHeader"
      />
    </div>
    <div
      className="HeaderDorothyPerkins-search"
      key="search"
    >
      <Connect(SearchBar)
        isDesktop={true}
      />
    </div>
    <div
      className="HeaderDorothyPerkins-account"
      key="account"
    >
      <Connect(AccountIcon)
        myAccountText="My account"
        popoverMenu={true}
        signOutText="Sign out"
      />
    </div>
    <div
      className="HeaderDorothyPerkins-wishlist"
      key="wishlist"
    >
      <Connect(WishlistHeaderLink) />
    </div>
    <div
      className="HeaderDorothyPerkins-shoppingCart"
      key="shoppingCart"
    >
      <Connect(ShoppingCart) />
    </div>
  </div>
  <div
    className="HeaderDorothyPerkins-navigation"
  >
    <Connect(TopNavMenuBig) />
  </div>
</div>
`;
